src/ripple/overlay/impl/Manifest.cpp

```diff
@@ -0,0 +1,247 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/main/Application.h>
+#include <ripple/app/peers/UniqueNodeList.h>
+#include <ripple/overlay/impl/Manifest.h>
+#include <ripple/protocol/RippleAddress.h>
+#include <ripple/protocol/Sign.h>
+#include <beast/http/rfc2616.h>
+#include <stdexcept>
+
+namespace ripple {
+
+static
+boost::optional<Manifest>
+unpackManifest(std::string s)
+{
+    STObject st(sfGeneric);
+    SerialIter sit(s.data(), s.size());
+    st.set(sit);
+    auto const mseq = get(st, sfSequence);
+    auto mpk  = get<AnyPublicKey>(st, sfPublicKey);
+    auto mspk = get<AnyPublicKey>(st, sfSigningPubKey);
+    if (! mseq || ! mpk || ! mspk)
+        return boost::optional<Manifest>();
```

Comment:
Here and anywhere else you return a default created `boost::optional<T>()`, you can use the simpler shorthand of `boost::none`.